tools/pi-appliance-update

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+UPDATE_TYPE="security"
+BOOT=0
+CLEAN="false"
+HELP=0
+
+while getopts "t:bch" option; do 
+ case "${option}" 
+ in 
+ t) UPDATE_TYPE=${OPTARG};; 
+ b) BOOT=1;; 
+ c) CLEAN="true";;
+ h) HELP=1;;
+ esac 
+done 
+
+if [ $HELP -eq 1 ]; then
+   echo "Usage: $0 [-b] [-c] [-t type]"
+   echo ""
+   echo " -b        : boot after update if necessary."
+   echo " -c        : clean unused packages after update."
+   echo " -t <type> : Either update 'security' packages or 'updates'."
+   exit 0
+fi
+
+LOGFILE=/var/log/privacyidea/update-${UPDATE_TYPE}.log
+
+function log {
+	echo $1 >> $LOGFILE
+}
+
+function do_update_security {
+	log "### Downloading packages."
+	aptitude update >> $LOGFILE
+	log "### Starting update."
+	aptitude safe-upgrade -o Aptitude::Delete-Unused=${CLEAN} --assume-yes --target-release `lsb_release -cs`-security >> $LOGFILE
+}
+
+function do_update_updates {
+	log "### Downloading packages."
+	aptitude update >> $LOGFILE
+	log "### Starting update."
+	aptitude full-upgrade -o Aptitude::Delete-Unused=${CLEAN} --assume-yes --target-release `lsb_release -cs`-updates >> $LOGFILE
+}
+
+
+date > $LOGFILE
```

Comment:
Hmm, so we overwrite the logfile with each update. Just want to make sure this is intentional?

Reply:
Yes, I am aware of this. 
This might not be optimal.
Otherwise we would have to rotate the logfile. We might want to do this in the long run...

Reply:
aptitude itself provides a logging facility with `--log-file=<file>` and `--log-level`

Reply:
Thanks for the hint. The question is, which information we want to save for a potential later debugging.

    aptitude update --log-file=apt.log --log-level=info

Will write a logfile with this content:

<pre>
2018-03-19 11:35:55 [140548281358208] ../../../../src/generic/util/temp.cc:207 INFO aptitude.temp - Initialized the temporary file module using the base directory /tmp/aptitude-cornelius.27940:iBWVYl
2018-03-19 11:35:55 [140548281358208] ../../../../src/generic/apt/apt.cc:399 INFO aptitude.apt.globals - Loading apt cache.
2018-03-19 11:35:56 [140548281358208] ../../../../src/generic/apt/download_queue.cc:866 INFO aptitude.downloadQueue - Shutting down the download queue.
2018-03-19 11:35:56 [140548281358208] ../../../../src/generic/apt/download_queue.cc:899 INFO aptitude.downloadQueue - The download queue is now shut down.
2018-03-19 11:35:56 [140548281358208] ../../../../src/generic/apt/apt.cc:332 INFO aptitude.apt.globals - Closing apt cache.
2018-03-19 11:36:02 [139897089857408] ../../../../src/generic/util/temp.cc:207 INFO aptitude.temp - Initialized the temporary file module using the base directory /tmp/aptitude-root.27970:NW8YZu
2018-03-19 11:36:02 [139897089857408] ../../../../src/generic/apt/apt.cc:399 INFO aptitude.apt.globals - Loading apt cache.
2018-03-19 11:36:06 [139897089857408] ../../../../src/generic/apt/apt.cc:332 INFO aptitude.apt.globals - Closing apt cache.
2018-03-19 11:36:12 [139897089857408] ../../../../src/generic/apt/apt.cc:399 INFO aptitude.apt.globals - Loading apt cache.
2018-03-19 11:36:14 [139897089857408] ../../../../src/generic/apt/download_queue.cc:866 INFO aptitude.downloadQueue - Shutting down the download queue.
2018-03-19 11:36:14 [139897089857408] ../../../../src/generic/apt/download_queue.cc:899 INFO aptitude.downloadQueue - The download queue is now shut down.
2018-03-19 11:36:14 [139897089857408] ../../../../src/generic/apt/apt.cc:332 INFO aptitude.apt.globals - Closing apt cache.
</pre>

We would be missing this content:

<pre>
Treffer http://de.archive.ubuntu.com/ubuntu xenial InRelease                                                                 
Holen: 1 http://security.ubuntu.com/ubuntu xenial-security InRelease [102 kB]                                                                        
Ign http://dl.google.com/linux/earth/deb stable InRelease                                                                            
Holen: 2 http://de.archive.ubuntu.com/ubuntu xenial-updates InRelease [102 kB]                               
Treffer http://dl.google.com/linux/chrome/deb stable Release                                                                                 
Treffer http://dl.google.com/linux/earth/deb stable Release                                                           
Treffer http://ppa.launchpad.net/privacyidea/privacyidea-dev/ubuntu xenial InRelease                                                                         
Treffer https://lancelot.netknights.it/apt/stable xenial InRelease                                                                                                      
Treffer https://repo.skype.com/deb stable InRelease                                                                                                                     
Ign http://download.opensuse.org/repositories/isv:/ownCloud:/desktop/Ubuntu_16.04  InRelease                                           
Holen: 3 http://de.archive.ubuntu.com/ubuntu xenial-backports InRelease [102 kB]                      
Treffer http://download.opensuse.org/repositories/isv:/ownCloud:/desktop/Ubuntu_16.04  Release             
Treffer https://repos.codelite.org/ubuntu xenial InRelease                           
Treffer http://ppa.launchpad.net/privacyidea/privacyidea/ubuntu xenial InRelease
Treffer https://downloads.iridiumbrowser.de/deb stable InRelease
Holen: 4 http://security.ubuntu.com/ubuntu xenial-security/main amd64 DEP-11 Metadata [67,6 kB]
Holen: 5 http://security.ubuntu.com/ubuntu xenial-security/main DEP-11 64x64 Icons [72,2 kB]
Holen: 6 http://security.ubuntu.com/ubuntu xenial-security/universe amd64 DEP-11 Metadata [51,4 kB]
Holen: 7 http://security.ubuntu.com/ubuntu xenial-security/universe DEP-11 64x64 Icons [85,1 kB]
Holen: 8 http://de.archive.ubuntu.com/ubuntu xenial-updates/main amd64 DEP-11 Metadata [317 kB]
Holen: 9 http://de.archive.ubuntu.com/ubuntu xenial-updates/main DEP-11 64x64 Icons [226 kB]
Holen: 10 http://de.archive.ubuntu.com/ubuntu xenial-updates/universe amd64 DEP-11 Metadata [190 kB]
Holen: 11 http://de.archive.ubuntu.com/ubuntu xenial-updates/universe DEP-11 64x64 Icons [272 kB]
Holen: 12 http://de.archive.ubuntu.com/ubuntu xenial-updates/multiverse amd64 DEP-11 Metadata [5.896 B]
Holen: 13 http://de.archive.ubuntu.com/ubuntu xenial-backports/main amd64 DEP-11 Metadata [3.328 B]
Holen: 14 http://de.archive.ubuntu.com/ubuntu xenial-backports/universe amd64 DEP-11 Metadata [5.088 B]
1.602 kB wurden in 1 s heruntergeladen (1.404 kB/s)   
</pre>